pkg/kv/kvserver/kvstorage/destroy_test.go

```diff
@@ -104,39 +130,80 @@ func (r *replicaInfo) createRaftState(ctx context.Context, t *testing.T, w stora
 	}
 }
 
-func (r *replicaInfo) createStateMachine(ctx context.Context, t *testing.T, rw storage.ReadWriter) {
+func (r *replicaInfo) createStateMachine(ctx context.Context, t *testing.T, w storage.Writer) {
 	sl := MakeStateLoader(r.id.RangeID)
-	require.NoError(t, sl.SetRangeTombstone(ctx, rw, kvserverpb.RangeTombstone{
+	require.NoError(t, sl.SetRangeTombstone(ctx, w, kvserverpb.RangeTombstone{
 		NextReplicaID: r.id.ReplicaID,
 	}))
-	require.NoError(t, sl.SetRaftReplicaID(ctx, rw, r.id.ReplicaID))
+	require.NoError(t, sl.SetRaftReplicaID(ctx, w, r.id.ReplicaID))
 	// TODO(pav-kv): figure out whether LastReplicaGCTimestamp should be in the
 	// log or state engine.
 	require.NoError(t, storage.MVCCBlindPutProto(
-		ctx, rw,
+		ctx, w,
 		keys.RangeLastReplicaGCTimestampKey(r.id.RangeID),
 		hlc.Timestamp{}, /* timestamp */
 		&hlc.Timestamp{WallTime: 12345678},
 		storage.MVCCWriteOptions{},
 	))
-	createRangeData(t, rw, r.keys)
+	createRangeData(t, w, r.keys)
 }
 
-func createRangeData(t *testing.T, rw storage.ReadWriter, span roachpb.RSpan) {
+func createRangeData(t *testing.T, w storage.Writer, span roachpb.RSpan) {
 	ts := hlc.Timestamp{WallTime: 1}
 	for _, k := range []roachpb.Key{
 		keys.RangeDescriptorKey(span.Key),   // system
 		span.Key.AsRawKey(),                 // user
 		roachpb.Key(span.EndKey).Prevish(2), // user
 	} {
 		// Put something under the system or user key.
-		require.NoError(t, rw.PutMVCC(
+		require.NoError(t, w.PutMVCC(
 			storage.MVCCKey{Key: k, Timestamp: ts}, storage.MVCCValue{},
 		))
 		// Put something under the corresponding lock key.
 		ek, _ := storage.LockTableKey{
 			Key: k, Strength: lock.Intent, TxnUUID: uuid.UUID{},
 		}.ToEngineKey(nil)
-		require.NoError(t, rw.PutEngineKey(ek, nil))
+		require.NoError(t, w.PutEngineKey(ek, nil))
+	}
+}
+
+type testBatch struct {
```

Comment:
@arulajmani something like that will be needed in #97617 (but slightly more general - `raftBatch` will be created lazily).